website/build.gradle

```diff
@@ -236,33 +231,24 @@ def buildContentDir(name) {
 }
 
 task serveWebsite(type: Exec) {
-  dependsOn setupDockerContainer, setupBuildDir
+  dependsOn setupDockerContainer
   finalizedBy stopAndRemoveDockerContainer
-  inputs.files 'Gemfile.lock', '_config.yml'
-  inputs.dir 'src'
-  outputs.dir "$buildDir/.sass-cache"
-  outputs.dir buildContentDir('local')
   commandLine 'docker', 'exec',
     "${->setupDockerContainer.containerId()}", '/bin/bash', '-c',
-    """cd $dockerWorkDir/build/website && \
-      bundle exec jekyll serve \
-      --config $dockerWorkDir/website/_config.yml,/tmp/_config_branch_repo.yml \
-      --incremental \
-      --source $dockerWorkDir/website/src \
-      --host 0.0.0.0
+    """cd $dockerSourceDir && \
+      yarn develop \
+      --bind="0.0.0.0" \
+      --config $dockerSourceDir/site/config.toml,/tmp/_config_branch_repo.toml
       """
 }
 
 task testWebsite(type: Exec) {
   // dependsOn setupDockerContainer, 'buildWebsite'
   finalizedBy stopAndRemoveDockerContainer
 
-  inputs.files "$buildDir/Rakefile"
-  inputs.dir buildContentDir('local')
   commandLine 'docker', 'exec',
     "${->setupDockerContainer.containerId()}", '/bin/bash', '-c',
-    """cd $dockerWorkDir/build/website && \
-      bundle exec -- rake test disable_external=${findProperty('disableExternal') ?: true}"""
+    "$dockerSourceDir/check-links.sh $dockerBuildDir/generated-local-content"
```

Comment:
Does check-links.sh cover a comparable amount of testing as the previous rake test?

Reply:
Jekyll testing is backed by Ruby with Rake. Unfortunately, we don't have the same thing officially in Hugo. Although the customization cannot be compared to the Rake test, it covers the most edge cases of broken links with various status codes.